pages/projects/[id]/edit.tsx

```diff
@@ -2,44 +2,52 @@ import React from 'react';
 import { useRouter } from 'next/dist/client/router';
-import { components } from '@/lib/types/api';
 import useData from '@/lib/hooks/useData';
+import { CustomerDTO, EmployeeDTO, ProjectDTO } from '@/lib/types/dto';
+import { customerEndpointURL, employeeEndpointURL, projectIdEndpointURL } from '@/lib/const';
+
+type Props = {
+    id: string;
+};
+
+function EditProjectForm({ id }: Props): JSX.Element {
+    const customerRequest = useData<CustomerDTO[]>(customerEndpointURL);
+    const employeesRequest = useData<EmployeeDTO[]>(employeeEndpointURL);
+    const projectRequest = useData<ProjectDTO>(projectIdEndpointURL(id));
 
-const Edit: NextPage = () => {
-    const {
-        data: customers,
-        isError: customerError,
-        isLoading: customersLoading,
-    } = useData<components['schemas']['CustomerDTO'][]>('customer');
-    const {
-        data: employees,
-        isError: employeeError,
-        isLoading: employeesLoading,
-    } = useData<components['schemas']['EmployeeDTO'][]>('employee');
-    const router = useRouter();
-    const { id } = router.query;
-    const { data: projects, isError, isLoading } = useData<components['schemas']['ProjectDTO']>(`project/${id}`);
     return (
         <Layout>
             <Heading fontWeight="black" margin="1rem 0rem">
                 Edit project
             </Heading>
-            {(customersLoading || employeesLoading) && <Loading></Loading>}
-            {(customerError || employeeError) && (
-                <ErrorAlert
-                    title="Error loading data"
-                    message="Could not load the required data from the server"
-                ></ErrorAlert>
+            {(customerRequest.isLoading || employeesRequest.isLoading) && <Loading />}
+            {(customerRequest.isError || employeesRequest.isError) && (
+                <ErrorAlert title="Error loading data" message="Could not load the required data from the server" />
+            )}
+            {projectRequest?.isLoading && <Loading />}
+            {projectRequest?.isError && (
+                <ErrorAlert title={projectRequest.isError.name} message={projectRequest.isError.name} />
+            )}
+            {customerRequest.data && employeesRequest.data && (
+                <ProjectForm
+                    customers={customerRequest.data}
+                    employees={employeesRequest.data}
+                    method="PUT"
+                    project={projectRequest?.data}
+                />
             )}
-            {isLoading && <Loading></Loading>}
-            {isError && <ErrorAlert title={isError.name} message={isError.name}></ErrorAlert>}
-            <ProjectForm customers={customers} employees={employees} method="PUT" project={projects}></ProjectForm>
         </Layout>
     );
+}
+
+const Edit: NextPage = () => {
+    const router = useRouter();
+    const { id } = router.query;
+    return id ? <EditProjectForm id={id[0]} /> : null;
```

Comment:
The correct way to handle the possibly missing `id` that's used in hooks is to conditionally render iff `id` is defined.

Reply:
Notice that `EditProjectForm` requires `id` to be defined. Thus the hooks work nicely.